### PR TITLE
Change defaultNetworkType to ovn-kubernetes

### PR DESCRIPTION
**Pull request by jluhrsen**
Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

### CODE REVIEW

**Comment by sadasu** [comment]
Is this being done for https://issues.redhat.com/browse/CORS-2085 ?

**Comment by jluhrsen** [comment]
> Is this being done for https://issues.redhat.com/browse/CORS-2085 ?

well, I suppose, but I didn't really know about this Epic. This PR is initially to see it work (running tests with it). I'm sure
there is more work to be added to this.

**Comment by patrickdillon** [comment]
Can you update the commit message(s) according to the format here: https://github.com/openshift/installer/blob/master/CONTRIBUTING.md#commit-message-format

**Comment by jluhrsen** [comment]
> Can you update the commit message(s) according to the format here: https://github.com/openshift/installer/blob/master/CONTRIBUTING.md#commit-message-format

updated. LMK if it's not good enough and I'll clean it up further.

**Comment by jluhrsen** [comment]
> Also it looks like some of the unit tests are failing. We can help with troubleshooting that if they prove problematic

should be good now.

**Comment by dgoodwin** [comment]
/lgtm

From TRT perspective we've roughly matched or exceeded parity with SDN jobs across the handful we have been monitoring.

**Comment by abhat** [comment]
@sadasu @patrickdillon - does this PR look good to go?

File: pkg/types/defaults/installconfig.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ var (
 	defaultServiceNetwork = ipnet.MustParseCIDR("172.30.0.0/16")
 	defaultClusterNetwork = ipnet.MustParseCIDR("10.128.0.0/14")
 	defaultHostPrefix     = 23
-	defaultNetworkType    = string(operv1.NetworkTypeOpenShiftSDN)
+	defaultNetworkType    = string(operv1.NetworkTypeOVNKubernetes)
```
Comment (sadasu) [inline pkg/types/defaults/installconfig.go:27]: Logic around https://github.com/openshift/installer/blob/master/pkg/types/defaults/installconfig.go#L50-L56 also needs to be cleaned up.

**Review by sadasu — COMMENTED** [review]
This PR can add kubebuilder default as OVNKubernetes. It was removed here https://github.com/openshift/installer/pull/5927/files#diff-ab4fb360ac4a8344c506c58e1b799894584bcd0daf8f1216c118b11cc079d375L279 when there was no default that was valid for all platforms.

File: pkg/types/validation/installconfig_test.go [hunk 2]
```diff
@@ -1277,28 +1277,6 @@ func TestValidateInstallConfig(t *testing.T) {
 			}(),
 			expectedError: `Invalid value: "IPv6": single-stack IPv6 is not supported for this platform`,
 		},
-		{
```
Comment (barbacbd) [inline pkg/types/validation/installconfig_test.go:1280]: Don't these tests still apply? It is still possible to select `OpenShiftSDN` as the network type, it is just not the default.

File: pkg/types/defaults/installconfig.go (hunk 1 above)
Comment (jluhrsen) [reply pkg/types/defaults/installconfig.go:27]: It's updated now @sadasu, I think it is much simpler now right, and just set to ovnk if ""

**Comment by jluhrsen** [comment]
> This PR can add kubebuilder default as OVNKubernetes. It was removed here https://github.com/openshift/installer/pull/5927/files#diff-ab4fb360ac4a8344c506c58e1b799894584bcd0daf8f1216c118b11cc079d375L279 when there was no default that was valid for all platforms.

I'm not exactly sure if I updated for this properly. I simply added a single line like:

`	// +kubebuilder:default=OVNKubernetes
`

please let me know if this is not correct, and what I need to do. I'll take care of it asap.

File: pkg/types/validation/installconfig_test.go (hunk 2 above)
Comment (jluhrsen) [reply pkg/types/validation/installconfig_test.go]: good catch. I added them back.

**Comment by abhat** [comment]
@jluhrsen you need to change the expected installconfigs.yml to add the default tag in the yaml too.

File: data/data/install.openshift.io_installconfigs.yaml [hunk 3]
```diff
@@ -1478,8 +1478,7 @@ spec:
                 type: array
               networkType:
                 description: NetworkType is the type of network to install. The default
```
Comment (abhat) [inline data/data/install.openshift.io_installconfigs.yaml:1480]: Need to add `default: OVNKubernetes` here if we will add kubebuilder validation.

Comment (jluhrsen) [reply data/data/install.openshift.io_installconfigs.yaml:1481]: thank you. updated.

**Comment by sadasu** [comment]
> > This PR can add kubebuilder default as OVNKubernetes. It was removed here https://github.com/openshift/installer/pull/5927/files#diff-ab4fb360ac4a8344c506c58e1b799894584bcd0daf8f1216c118b11cc079d375L279 when there was no default that was valid for all platforms.
> 
> I'm not exactly sure if I updated for this properly. I simply added a single line like:
> 
> ` // +kubebuilder:default=OVNKubernetes `
> 
> please let me know if this is not correct, and what I need to do. I'll take care of it asap.

Right before this comment https://github.com/openshift/installer/blob/master/pkg/types/installconfig.go#L283, please add 
 // +kubebuilder:default=OVNKubernetes , as you correctly pointed out earlier.

File: pkg/types/conversion/installconfig.go [hunk 4]
```diff
@@ -78,8 +78,8 @@ func convertNetworking(config *types.InstallConfig) {
 
 	// Recognize the default network plugin name regardless of capitalization, for
 	// backward compatibility
-	if strings.ToLower(netconf.NetworkType) == strings.ToLower(string(operv1.NetworkTypeOpenShiftSDN)) {
-		netconf.NetworkType = string(operv1.NetworkTypeOpenShiftSDN)
+	if strings.ToLower(netconf.NetworkType) == strings.ToLower(string(operv1.NetworkTypeOVNKubernetes)) {
+		netconf.NetworkType = string(operv1.NetworkTypeOVNKubernetes)
```
Comment (danwinship) [inline pkg/types/conversion/installconfig.go:82]: This should not be changed. Well, the comment should be changed to not say "default", but the purpose of this code is not "whatever the default networkType value is, we should recognize it case-insensitively", it's "the correct value of this field used to be `OpenshiftSDN` but then we fixed it to be `OpenShiftSDN`, but we need to continue matching both values for backward-compatibility with (very) old install-configs.

File: pkg/types/conversion/installconfig_test.go [hunk 5]
```diff
@@ -149,25 +149,6 @@ func TestConvertInstallConfig(t *testing.T) {
 			},
 			expectedError: "no version was provided",
 		},
-		{
-			name: "deprecated OpenShiftSDN spelling",
```
Comment (danwinship) [inline pkg/types/conversion/installconfig_test.go:153]: and likewise this should be preserved

File: pkg/types/defaults/installconfig.go [hunk 6]
```diff
@@ -48,11 +48,7 @@ func SetInstallConfigDefaults(c *types.InstallConfig) {
 		}
 	}
 	if c.Networking.NetworkType == "" {
-		if c.IsOKD() || c.IsSingleNodeOpenShift() {
-			c.Networking.NetworkType = string(operv1.NetworkTypeOVNKubernetes)
-		} else {
-			c.Networking.NetworkType = defaultNetworkType
-		}
+		c.Networking.NetworkType = defaultNetworkType
```
Comment (danwinship) [inline pkg/types/defaults/installconfig.go:51]: is this even needed? with the `kubebuilder:default`, won't it always have gotten filled in if the user didn't specify it?

**Review by danwinship — CHANGES_REQUESTED** [review]

File: pkg/types/defaults/installconfig.go (hunk 6 above)
Comment (jluhrsen) [reply pkg/types/defaults/installconfig.go:51]: is there no other way for c.Networking.NetworkType == ""

I don't know. this doesn't hurt, but if you know for certain then I can remove it.

File: pkg/types/conversion/installconfig_test.go (hunk 5 above)
Comment (jluhrsen) [reply pkg/types/conversion/installconfig_test.go]: fixed

File: pkg/types/conversion/installconfig.go (hunk 4 above)
Comment (jluhrsen) [reply pkg/types/conversion/installconfig.go]: fixed

**Comment by sdodson** [comment]
/retest-required
4.12.0-0.ci-2022-07-28-142749 with fixes for the acl issue has now been accepted.

**Comment by patrickdillon** [comment]
Looking at the history of e2e-vsphere runs, we have had several successful installs and they are all failing with:

>   [sig-network][Feature:Network Policy Audit logging] when using openshift ovn-kubernetes should ensure acl logs are created and correct [Suite:openshift/conformance/parallel] 

[example](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_installer/6014/pull-ci-openshift-installer-master-e2e-vsphere/1552651967061299200)

**Comment by sdodson** [comment]
@patrickdillon I think 4.12.0-0.ci-2022-07-28-142749 included https://github.com/openshift/cluster-policy-controller/pull/82
I was hoping to see a successful vsphere run after that fix landed but feel free to override as you wish.

**Comment by patrickdillon** [comment]
> @patrickdillon I think 4.12.0-0.ci-2022-07-28-142749 included [openshift/cluster-policy-controller#82](https://github.com/openshift/cluster-policy-controller/pull/82) I was hoping to see a successful vsphere run after that fix landed but feel free to override as you wish.

doh I see your earlier comment now. thanks for clarifying.

**Comment by jluhrsen** [comment]
> @sdodson, @patrickdillon: are we ready to lgtm again or do we think the network policy audit logging failure on vsphere should be gating this PR?

vsphere passed

File: pkg/types/defaults/installconfig.go (hunk 6 above)
Comment (danwinship) [reply pkg/types/defaults/installconfig.go:51]: yeah, dunno. it doesn't hurt

**Pull request merged**